### DISCUSSION TITLE
Configurable Exit OR Completion hook?

### DISCUSSION THREAD

**Discussion by nickhudkins** [kamilkisiela/graphql-inspector#1474] [General] [answered]
Would you be open to adding the ability to fail on different levels of Criticality?

What I am imagining is something like:

`graphql-inspector diff old.graphql new.graphql --failureLevel DANGEROUS`
`graphql-inspector diff old.graphql new.graphql --failureLevel NON_BREAKING`

This would allow consumers to choose when to fail CI.

I understand that this COULD be implemented using `rules` and "upgrading" these failures to `CriticalityLevel.BREAKING`, but this fundamentally "hides" the actual criticality.

The desire here is simply to be able to cause the process to exit(1), based on a failureLevel.

Optionally, we could implement `onComplete` so that we aren't supporting such a specific use-case and instead allowing people to handle the finished diffing operation:

`graphql-inspector diff old.graphql new.graphql --onComplete myCustomCompletion`

```
module.exports = (changes: Changes[], oldSchema: Schema, newSchema: Schema) => {
	const hasDangerousChanges = changes.filter(isDangerous).length > 0
    if (hasDangerousChanges) {
		console.error(`[Dangerous Changes Detected]: Exiting...`)
        process.exit(1)
	}
}
```

and implement the default behavior of `onComplete` as:

https://github.com/kamilkisiela/graphql-inspector/blob/master/packages/commands/diff/src/index.ts#L114-L125

I plan on going down the `onComplete` path in a fork, but wanted to see if this was something of interest to anyone else

**Comment by kamilkisiela** [comment, marked as answer]
MERGED 🚀

**Reply by nickhudkins** [reply]
Awesome! Thanks so much. I'll get a PR in to update the docs in the next day or two.

**Reply by ides15** [reply]
Hey @nickhudkins probably a long shot since this was a while ago but did you ever get a PR in for docs for this feature? Thanks for the change!
